src/services/auth/service.ts

```diff
@@ -45,37 +45,28 @@ export class AuthService {
     destination.searchParams.set('url', encodeURIComponent(redirectionUrl));
     const link = destination.toString();
 
-    const lang = member.lang;
-
-    const translated = this.mailerService.translate(lang);
-    const subject = translated(MAIL.SIGN_UP_TITLE);
-    const greetingsAndSignupText = `${translated(MAIL.GREETINGS)} ${translated(MAIL.SIGN_UP_TEXT)}`;
-    const html = `
-    ${this.mailerService.buildText(greetingsAndSignupText)}
-    ${this.mailerService.buildButton(link, translated(MAIL.SIGN_UP_BUTTON_TEXT))}
-    ${this.mailerService.buildText(
-      translated(MAIL.USER_AGREEMENTS_MAIL_TEXT, {
-        signUpButtonText: translated(MAIL.SIGN_UP_BUTTON_TEXT),
-        graaspLandingPageOrigin: GRAASP_LANDING_PAGE_ORIGIN,
-      }),
-      // Add margin top of -15px to remove 15px margin bottom of the button.
-      { 'text-align': 'center', 'font-size': '10px', 'margin-top': '-15px' },
-    )}
-    ${this.mailerService.buildText(translated(MAIL.SIGN_UP_NOT_REQUESTED))}`;
-
-    const footer = this.mailerService.buildFooter(lang);
+    const mail = new MailBuilder({
+      subject: { text: MAIL.SIGN_UP_TITLE },
+      lang: member.lang,
+    })
+      .addText(MAIL.GREETINGS)
+      .addText(MAIL.SIGN_UP_TEXT)
+      .addButton(MAIL.SIGN_UP_BUTTON_TEXT, link)
+      .addUserAgreement(MAIL.SIGN_UP_BUTTON_TEXT)
+      .addSignUpNotRequested()
+      .build();
 
     // don't wait for mailerService's response; log error and link if it fails.
     this.mailerService
-      .sendEmail(subject, member.email, link, html, footer)
+      .send(mail, member.email)
       .catch((err) => this.log.warn(err, `mailerService failed. link: ${link}`));
   };
 
   generateLoginLinkAndEmailIt = async (
     member: Member,
     options: { challenge?: string; lang?: string; url?: string } = {},
```

Comment:
If `lang` is not used in the function, remove it from the options and the call site.